task: Fix formatting of ./sync files and add CI job for ruff

### Component

_No response_

### Task Description

Running ruff in the root of the repo shows formatting issues within the sync project. Would be good to clean these up to remove the noise and also add a CI step that validates these files if changed.

```bash
❯ ruff check .
sync/infrahub-sync/infrahub_sync/adapters/infrahub.py:1:1: I001 [*] Import block is un-sorted or un-formatted
sync/infrahub-sync/infrahub_sync/adapters/nautobot.py:1:1: I001 [*] Import block is un-sorted or un-formatted
sync/infrahub-sync/infrahub_sync/adapters/netbox.py:1:1: I001 [*] Import block is un-sorted or un-formatted
sync/potenda/potenda/__init__.py:1:1: I001 [*] Import block is un-sorted or un-formatted
Found 4 errors.
[*] 4 fixable with the `--fix` option.
```

--- a/sync/infrahub-sync/infrahub_sync/adapters/infrahub.py
+++ b/sync/infrahub-sync/infrahub_sync/adapters/infrahub.py
@@ -1,7 +1,6 @@
 import copy
 from typing import Any, Dict, Mapping
 
-from diffsync import DiffSync, DiffSyncModel
 from infrahub_sdk import (
     Config,
     InfrahubClientSync,
@@ -12,6 +11,7 @@ from infrahub_sdk import (
 )
 from infrahub_sdk.utils import compare_lists
 
+from diffsync import DiffSync, DiffSyncModel
 from infrahub_sync import DiffSyncMixin, DiffSyncModelMixin, SyncAdapter, SyncConfig
 from infrahub_sync.generator import has_field
 

--- a/sync/infrahub-sync/infrahub_sync/adapters/nautobot.py
+++ b/sync/infrahub-sync/infrahub_sync/adapters/nautobot.py
@@ -4,8 +4,8 @@ import os
 from typing import TYPE_CHECKING, Any, Dict
 
 import pynautobot
-from diffsync import DiffSync, DiffSyncModel
 
+from diffsync import DiffSync, DiffSyncModel
 from infrahub_sync import (
     DiffSyncMixin,
     DiffSyncModelMixin,

--- a/sync/infrahub-sync/infrahub_sync/adapters/netbox.py
+++ b/sync/infrahub-sync/infrahub_sync/adapters/netbox.py
@@ -4,8 +4,8 @@ import os
 from typing import TYPE_CHECKING, Any, Dict
 
 import pynetbox
-from diffsync import DiffSync, DiffSyncModel
 
+from diffsync import DiffSync, DiffSyncModel
 from infrahub_sync import (
     DiffSyncMixin,
     DiffSyncModelMixin,

--- a/sync/potenda/potenda/__init__.py
+++ b/sync/potenda/potenda/__init__.py
@@ -1,11 +1,11 @@
 from typing import List, Optional
 
+from tqdm import tqdm
+
 from diffsync import DiffSync
 from diffsync.diff import Diff
 from diffsync.enum import DiffSyncFlags
 from diffsync.logging import enable_console_logging
-from tqdm import tqdm
-
 from infrahub_sync import SyncInstance
 
 

--- a/sync/pyproject.toml
+++ b/sync/pyproject.toml
@@ -188,7 +188,7 @@ skip-magic-trailing-comma = false
 line-ending = "auto"
 
 [tool.ruff.lint.isort]
-known-first-party = ["infrahub_sync", "potenda"]
+known-first-party = ["diffsync", "infrahub_sync", "potenda"]
 
 [tool.ruff.lint.pycodestyle]
 max-line-length = 150

